File: lib/models/payment_method.dart
enum PaymentType { debitCard, creditCard, cash, wallet, bankTransfer, bizum }

class PaymentMethod {
  final String id;
  final String userId;
  final String name;
  final PaymentType type;
  final String? issuer;
  final bool isDefault;
  final bool isActive;
  final DateTime createdAt;
  final DateTime updatedAt;

  PaymentMethod({
    required this.id,
    required this.userId,
    required this.name,
    required this.type,
    required this.issuer,
    required this.isDefault,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });
}

File: backend/src/main/java/com/walletapp/demo/entity/PaymentMethod.java
package com.walletapp.demo.entity;

import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import com.walletapp.demo.enums.PaymentType;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "payment_methods", indexes = {
        @Index(name = "idx_payment_method_user_id", columnList = "created_by_id"),
        @Index(name = "idx_payment_method_issuer", columnList = "issuer"),

})
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
public class PaymentMethod {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "payment_method_id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "created_by_id", nullable = false)
    private User user;
    private String alias;

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_type")
    private PaymentType type;

    @Column(nullable = false)
    private String issuer;

    @Column(name = "is_default")
    private boolean isDefault;

    @Column(name = "is_active")
    private boolean isActive;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
